### ISSUE TITLE
No longer works in macOS Catalina 

### ISSUE THREAD

**Issue by craiggrummitt** [larkob/GB2MIDI#2]
Thanks for putting this tool together. The online tool saved the day for me, as the app appears to not work in macOS Catalina (perhaps it's not 64-bit?). Thought you might like to know.

**Comment by destroybabylon** [comment]
I would also love to see an update for Catalina, as this tool is super useful. Thanks

**Comment by kameniko** [comment]
Yes, please! It is really useful!

**Comment by scotttroyer** [comment]
Hi Lars! People love your app. I [blogged about it](https://scotttroyer.com/2014/05/export-midi-from-garageband/) a while ago after finding it really useful myself. I get requests probably once a week now asking for Catalina support. Just thought I'd let you know there appears to be demand. Thanks!

**Comment by hessismore** [comment]
Hey, great app, super useful.
I was hoping to have it working on Catalina (64-bit).
Meanwhile are there any alternatives?

**Comment by boredjames** [comment]
PLEASE PLEASE PLEASE UPDATE IT FOR CATALINA

**Comment by AlanYost** [comment]
Works really well on Catalina for me - Version: 10.15.5 (19F96). I used the web browser option.  https://larkob.github.io/GB2MIDI/index.html

**Comment by jonblower** [comment]
If you're happy to use the Terminal, you can just use the underlying Perl script. This works fine in Catalina, as far as I can tell.

- Download the Perl script `gb2midi.pl` from this GitHub repo (direct link). You could put it anywhere, but maybe it's most useful in the directory where GB puts its loops
- Create the `.aif` loop as instructed in the README
- To extract the midi, open a Terminal window and enter `perl gb2midi.pl my_loop.aif my_loop.mid`

(This assumes that the `.aif` file is in the same directory as the Perl script.)

**Comment by Sford4** [comment]
Very useful app, Catalina would be appreciated!

**Comment by sokobania** [comment]
I agree with **jonblower**. The current version __can work with Catalina__. More precisely, here is what I did:

Note that `~` refers to my home directory `/Users/jack`.

- **Safari**: Download [GB2MIDI.zip](https://github.com/larkob/GB2MIDI/blob/master/GB2MIDI.zip)
- **Finder**: Expand the zip file : double click `/Users/jack/Downloads/GB2MIDI.zip` to create `/Users/jack/Downloads/GB2MIDI.app`
- **Finder**: Move the app : move `/Users/jack/Downloads/GB2MIDI.app` into `/Applications/GB2MIDI.app`
- **Finder**: Create the destination directory for your MIDI files, e.g. `/Users/jack/Music/MIDI`
- **Garageband**: Create the `.aif` loop as instructed in the README
- **Terminal**: Type : `ls -ltr ~/"Library/Audio/Apple Loops/User Loops/SingleFiles/"`
  - => this will list your loops (with the most recent displayed last)
- Since the Perl script `gb2midi.pl` (embedded inside the application) is executable and begins with the right shebang, you can then type (on one line):
- **Terminal**: Type: `/Applications/GB2MIDI.app/Contents/Resources/Scripts/gb2midi.pl ~/"Library/Audio/Apple Loops/User Loops/SingleFiles/myPrettyLoop.aif" ~/"Music/MIDI/myPrettyLoop.mid"`
  - => you get the MIDI file from your loop in your MIDI directory !

Note that, since the directory names `Apple Loops` and `User Loops` contain a space character, 

- either they must be surrounded by quote characters:
  -  `~/"Library/Audio/Apple Loops/User Loops/SingleFiles/myPrettyLoop.aif"`
- either the space character must be quoted with a backslash character
  - `~/Library/Audio/Apple\ Loops/User\ Loops/SingleFiles/myPrettyLoop.aif`.

Note that the whole app itself `GB2MIDI.app` is not really needed (since it does not run under Catalina). 
If you prefer, you can just download the Perl script [gb2midi.pl](https://github.com/larkob/GB2MIDI/blob/master/GB2MIDI.app/Contents/Resources/Scripts/gb2midi.pl) as mentioned by **jonblower** and adapt the Terminal command.

**Comment by hariby** [comment]
The app didn't run on Big Sur 11.1 (20C69) either, 
> **“GB2MIDI” needs to be updated.** The developer of this app needs to update it to work with this version of macOS. Contact the developer for more information.

but the browser version still works well. 
https://larkob.github.io/GB2MIDI/index.html

**Comment by Brotuulaan** [comment]
Is this essentially an audio-to-midi converter? I had a friend play a piano part for one of my projects in Ableton, and we ran into the whole GrageBand/midi problem. The file is too complex for audio converters to handle (it takes so much work to fix the resulting file that we might as well program the midi file from scratch, and then it would still lack all the nuance of a live performance). I found this listed on a number of sites, but I keep seeing the .aif starting point and wondering what makes this any different from other audio converters out there.

**Comment by jonblower** [comment]
@Brotuulaan - I don't think this is audio-to-midi. The .aif files in GarageBand can contain audio or midi loops, but GB doesn't provide a way of easily extracting midi information to a standard midi file. That's what this tool is about. I assume any audio information in the loop would be ignored.

**Comment by scotttroyer** [comment]
@Brotuulaan @jonblower That is correct—GB2MIDI only extracts the MIDI data from the .aif file. It _cannot_ convert audio to MIDI. There are programs that can convert audio to MIDI, but results I've seen never quite capture all the nuances. Some polishing of the MIDI is always needed.
